### PR TITLE
Feature/issue 2 create endpoint historical prices

### PR DESCRIPTION
**Pull request by ajay567**
**Design**
Add the historical prices path in application.yml and the api token as an environment variable.
Create a new domain for Historical Prices (class: IexHistoricalPrices).
Create a new service for Historical Prices (interface: IexHistoricalClient)
Create mapping and methods for historical prices in IexService and IexRestController.

**Test Evidence**
Create a wiremock mapping for historical prices (wiremock: mapping-historicalPrice.json)
Create a test method (testGetHistoricalPrices) in IexRestControllerTest. Test historical prices using the wiremoch created.

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+iexApiToken = ${IEX_API_TOKEN}
```
Comment (grundygolem2) [inline src/main/resources/application.properties]: is there no way to put this in the yml?

Comment (ajay567) [reply src/main/resources/application.properties:1]: I just tried doing it in yml and it works. It is the exact same way.

Comment (ajay567) [reply src/main/resources/application.properties]: > is there no way to put this in the yml?
I just tried doing it in yml and it works. It is the exact same way.

Comment (grundygolem2) [reply src/main/resources/application.properties]: lets do that instead then. that way we only have one config file

Comment (ajay567) [reply src/main/resources/application.properties]: Made the changes and added it to the branch.

File: pom.xml [hunk 2]
```diff
@@ -180,7 +180,7 @@
 		<dependency>
 			<groupId>com.fasterxml.jackson.core</groupId>
 			<artifactId>jackson-databind</artifactId>
-			<version>2.9.10.3</version>
+			<version>2.9.10.5</version>
```
Comment (alex-power) [inline pom.xml:183]: Why was this upgraded?

File: src/main/java/org/galatea/starter/service/IexHistoricalClient.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import org.galatea.starter.domain.IexHistoricalPrices;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+/**
+ * A Feign Declarative REST Client to access endpoints from the Free and Open IEX API to get market
+ * data. See https://iextrading.com/developer/docs/
+ */
+@FeignClient(name = "IEXHIST", url = "${spring.rest.iexCloudPath}")
```
Comment (alex-power) [inline src/main/java/org/galatea/starter/service/IexHistoricalClient.java:13]: use camelCase for ```name```

**Review by grundygolem2 — APPROVED** [review]

File: src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java [hunk 4]
```diff
@@ -70,6 +70,26 @@ public void testGetLastTradedPrice() throws Exception {
...
+        .andExpect(jsonPath("$[0].close").value(new BigDecimal("116.59")))
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("117.49")))
+        .andExpect(jsonPath("$[0].low").value(new BigDecimal("116.22")))
+        .andExpect(jsonPath("$[0].open").value(new BigDecimal("116.57")))
+        .andExpect(jsonPath("$[0].symbol", is("AAPL")))
+        .andExpect(jsonPath("$[0].volume").value(new BigDecimal("46691331")))
+        .andExpect(jsonPath("$[0].date", is("2021-06-27")))
+        .andReturn();
+  }
+
```
Comment (alex-power) [inline src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java:92]: Add tests for different cases
* symbol, range, and date, not provided
* range and date not provided
* date not provided
* symbol not provided
* iex server failure/HTTP 500 error

File: src/main/java/org/galatea/starter/service/IexHistoricalClient.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.util.List;
+import org.galatea.starter.domain.IexHistoricalPrices;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+/**
+ * A Feign Declarative REST Client to access endpoints from the Free and Open IEX API to get market
+ * data. See https://iextrading.com/developer/docs/
```
Comment (alex-power) [inline src/main/java/org/galatea/starter/service/IexHistoricalClient.java:11]: this javadoc is copy/pasted, not applicable to this client?

File: src/main/java/org/galatea/starter/service/IexHistoricalClient.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ */
+@FeignClient(name = "IEXHIST", url = "${spring.rest.iexCloudPath}")
+public interface IexHistoricalClient {
+
+  /**
+   * Get the historical prices for each stock symbol passed in. See https://iextrading.com/developer/docs/#last.
+   * http://localhost:8080/iex/historicalPrices?symbol=MSFT&range=1m&date=
```
Comment (alex-power) [inline src/main/java/org/galatea/starter/service/IexHistoricalClient.java:18]: please clean up this javadoc as well

File: src/main/java/org/galatea/starter/service/IexHistoricalClient.java (hunk 5 above)
Comment (ajay567) [reply src/main/java/org/galatea/starter/service/IexHistoricalClient.java:11]: Yes. I copied it. Will make changes.

File: pom.xml (hunk 2 above)
Comment (ajay567) [reply pom.xml:183]: I can revert it back. Did not make this change manually.

File: src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java (hunk 4 above)
Comment (ajay567) [reply src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java:92]: Added the test cases on a different branch. Created a pull request for it.

**Pull request merged**